azure/services/scalesets/scalesets.go

```diff
@@ -517,6 +526,19 @@ func (s *Service) buildVMSSFromSpec(ctx context.Context, vmssSpec azure.ScaleSet
 		},
 	}
 
+	switch orchestrationMode {
+	case compute.OrchestrationModeUniform:
+		vmss.VirtualMachineScaleSetProperties.Overprovision = to.BoolPtr(false)
+		vmss.VirtualMachineScaleSetProperties.UpgradePolicy = &compute.UpgradePolicy{Mode: compute.UpgradeModeManual}
+	case compute.OrchestrationModeFlexible:
+		vmss.VirtualMachineScaleSetProperties.VirtualMachineProfile.NetworkProfile.NetworkAPIVersion =
+			compute.NetworkAPIVersionTwoZeroTwoZeroHyphenMinusOneOneHyphenMinusZeroOne
+		vmss.VirtualMachineScaleSetProperties.PlatformFaultDomainCount = to.Int32Ptr(1)
```

Comment:
So basically the VMSS Flex API exposes a new `PlatformFaultDomainCount` property that is a peer property to `Zones`, and we always want to ship a VMSS Flex config where those properties have the same value?

Reply:
Yes, that's my understanding: `PlatformFaultDomainCount` is required for `OrchestrationMode: Flexible` and shouldn't be greater than the number of zones.

Reply:
ACK

If we wanted to expose this to the user later will this current default enforcement be forward-compatible with such a change in the future? I think so.